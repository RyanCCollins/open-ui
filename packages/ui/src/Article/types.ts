import BoxProps from '../Box/types';

export interface ArticleProps extends BoxProps {
  content?: string;
  children?: JSX.Element;
}
