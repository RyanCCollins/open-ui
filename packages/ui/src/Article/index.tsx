import * as React from 'react';
import Component from './styles';
import { ArticleProps } from './types';
const Markdown = require('react-markdown');
require('github-markdown-css/github-markdown.css');

export default function Article({
  content,
  children,
  ...rest,
}: ArticleProps) {
  return (
    <Component {...rest} className="markdown-body">
      {content && typeof content === 'string' &&
        <Markdown source={content} />
      }
      {children}
    </Component>
  );
}
