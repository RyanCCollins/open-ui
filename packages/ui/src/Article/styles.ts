const styled = require('styled-components').default;
import {
  BoxStyles,
} from '../Box/styles';

export default styled.article`
  ${BoxStyles}
  box-shadow: 0 2px 4px 0 rgba(46, 61, 73, 0.2);
  border: 1px solid #dbe2e8;
`;
