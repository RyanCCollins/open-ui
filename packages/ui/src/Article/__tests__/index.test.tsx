import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import * as React from 'react';
import Article from '../';
import props from './__mocks__/articleMocks.mock';

describe('<Article />', () => {
  it('should render with default props', () => {
    const wrapper = shallow(
      <Article />,
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
  it('should render with default props and children', () => {
    const wrapper = shallow(
      <Article>
        ðŸ‘¶ðŸ‘¶ðŸ‘¶
      </Article>,
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
  it('should render with different props', () => {
    const wrapper = shallow(
      <Article {...props}>
        ðŸ‘¶ðŸ‘¶ðŸ‘¶
      </Article>,
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
  it('should render with content', () => {
    const wrapper = shallow(
      <Article {...props} content="# Awesome" />,
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
