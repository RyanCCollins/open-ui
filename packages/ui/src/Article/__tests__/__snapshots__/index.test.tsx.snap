// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Article /> should render with content 1`] = `
<styled.article
  backgroundColor="#f5f5f5"
  className="markdown-body"
  pad="medium"
  size="large"
>
  <ReactMarkdown
    containerTagName="div"
    source="# Awesome"
  />
</styled.article>
`;

exports[`<Article /> should render with default props 1`] = `
<styled.article
  className="markdown-body"
/>
`;

exports[`<Article /> should render with default props and children 1`] = `
<styled.article
  className="markdown-body"
>
  ğŸ‘¶ğŸ‘¶ğŸ‘¶
</styled.article>
`;

exports[`<Article /> should render with different props 1`] = `
<styled.article
  backgroundColor="#f5f5f5"
  className="markdown-body"
  pad="medium"
  size="large"
>
  ğŸ‘¶ğŸ‘¶ğŸ‘¶
</styled.article>
`;
